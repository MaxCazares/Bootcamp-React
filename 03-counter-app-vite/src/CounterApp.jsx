import { Fragment, useState } from "react"

export const CounterApp = ({value = 0}) => {

    const [counter, setCounter] = useState(value)

    const handleSubstract = () => setCounter(counter - 1)
    const handleReset = () => setCounter(value)
    const handleAdd = () => setCounter(counter + 1)

    return (
        <Fragment>
            <h1>Counter App</h1>
            <h2 data-testid='test-counter'>{counter}</h2>
            <button aria-label="btn-decrease" onClick={handleSubstract}>-1</button>
            <button aria-label="btn-reset" onClick={handleReset}>Reset</button>
            <button aria-label="btn-increase" onClick={handleAdd}>+1</button>
        </Fragment>
    )
}