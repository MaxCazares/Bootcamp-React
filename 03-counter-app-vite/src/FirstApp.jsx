import { Fragment } from "react"
import PropTypes from 'prop-types'

const person = {
    'Name': 'Max',
    'Age': 25
}

const getFactorialNumber = number => {
    var factorial = 1;
    for (let i = 1; i <= number; i++) 
        factorial *= i;

    return factorial
}

export const FirstApp = ({title = 'React', factorial = 1}) => {
    return (
    <Fragment>
        <h2>{title}'s First App</h2>
        <code>{JSON.stringify(person)}</code>
        <p>{getFactorialNumber(factorial)}</p>
    </Fragment>
    )
}
