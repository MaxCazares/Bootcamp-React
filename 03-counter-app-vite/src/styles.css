html, body{
    background-color: #21232a;
    color: white;
    padding: 70px;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
}

button{
    font-size: 1.3rem;
    padding: 10px;
    margin-right: 10px;
}