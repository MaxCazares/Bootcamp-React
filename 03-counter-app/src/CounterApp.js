import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);

    const incrementarValue = () => {
        setCounter(counter + 1);
        // setCounter( c => c + 1 );
    }

    const deincrementarValue = () => setCounter(counter - 1);

    const resetValue = () => setCounter(start);

    return (
        <Fragment>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>

            <button onClick={deincrementarValue}>-1</button>
            <button onClick={resetValue}>Reset</button>
            <button onClick={incrementarValue}>+1</button>
        </Fragment>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 10
}

export default CounterApp;