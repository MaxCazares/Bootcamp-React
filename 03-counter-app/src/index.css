html, body{
    background-color: #21232A;
    color: #fff;
    font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
    font-size: 13.rem;
    padding: 70px;
}

button{
    font-size: 1.3rem;
    padding: 10px;
    margin-right: 10px;
}