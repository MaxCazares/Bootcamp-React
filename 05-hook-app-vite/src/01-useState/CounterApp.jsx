import { Fragment, useState } from "react"

export const CounterApp = () => {

    const startValue = {
        counter1: 10,
        counter2: 20,
        counter3: 30,
    }
    const [counter, setCounter] = useState(startValue)
    const {counter1, counter2, counter3} = counter        

    return (
        <Fragment >
            <h1 className="w-1/6 mx-auto m-5 bg-blue-300 shadow-2xl rounded-2xl p-3 text-4xl text-center">Counter1: {counter1}</h1>
            <h1 className="w-1/6 mx-auto m-5 bg-blue-300 shadow-2xl rounded-2xl p-3 text-4xl text-center">Counter2: {counter2}</h1>
            <h1 className="w-1/6 mx-auto m-5 bg-blue-300 shadow-2xl rounded-2xl p-3 text-4xl text-center">Counter3: {counter3}</h1>

            <div className="flex mx-auto w-1/5 space-x-10">
                <button
                    className="bg-red-400 mx-auto rounded-2xl shadow-2xl px-3 py-2 text-2xl text-center"
                    onClick={() => setCounter({
                        ...counter,
                        counter1: counter1 + 1,
                    })}> 
                +1
                </button>
            </div>
        </Fragment>
    )
}
