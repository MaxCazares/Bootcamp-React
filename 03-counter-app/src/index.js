import React from 'react';
import ReactDOM from 'react-dom';
// import PrimeraApp from './PrimeraApp';
import CounterApp from './CounterApp';
import './index.css'

const root = document.querySelector('#root');

// ReactDOM.render(<PrimeraApp saludo='Hola Mundo' />, root);
ReactDOM.render(<CounterApp value={123} />, root);