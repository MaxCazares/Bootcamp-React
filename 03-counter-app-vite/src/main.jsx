import React from "react";
import ReactDOM from "react-dom/client";

import './styles.css'

import {App} from './HelloWorldApp.jsx'     
import {FirstApp} from './FirstApp.jsx'
import {CounterApp} from './CounterApp.jsx'


ReactDOM.createRoot((document.getElementById('root'))).render(
    <React.StrictMode>
        {/* <App/> */}
        {/* <FirstApp title="Max" factorial={10}/> */}
        <CounterApp value={123}/>
    </React.StrictMode>
)

