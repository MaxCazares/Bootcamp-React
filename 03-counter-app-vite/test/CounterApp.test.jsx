import { render, screen, fireEvent } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';

describe('Pruebas en <CounterApp />', () => {

    const value = 100

    test('Debe de hacer match con el snapshot', () => {
        const {container} = render(<CounterApp value={value}/>)
        expect(container).toMatchSnapshot()
    })

    test('Debe mostrar el valor inicial de 100', () => {        
        render(<CounterApp value={value}/>)
        expect(screen.getByTestId('test-counter').innerHTML).toContain(value.toString())
        expect(screen.getByText(value)).toBeTruthy() 
    })

    test('debe de incrementar con el boton de +1', () => {
        render(<CounterApp value={value}/>)
        fireEvent.click(screen.getByText('+1'))
        expect(screen.getByText(value + 1)).toBeTruthy()

    })

    test('debe de incrementar con el boton de -1', () => {
        render(<CounterApp value={value}/>)
        fireEvent.click(screen.getByText('-1'))
        expect(screen.getByText(value - 1)).toBeTruthy()

    })

    test('debe de funcionar el boton de reset', () => {
        render(<CounterApp value={value}/>)
        fireEvent.click(screen.getByRole('button', {name: 'btn-increase'}))
        fireEvent.click(screen.getByRole('button', {name: 'btn-decrease'}))
        fireEvent.click(screen.getByRole('button', {name: 'btn-increase'}))
        fireEvent.click(screen.getByRole('button', {name: 'btn-decrease'}))
        fireEvent.click(screen.getByRole('button', {name: 'btn-decrease'}))
        screen.debug()
        fireEvent.click(screen.getByRole('button', {name: 'btn-reset'}))

        expect(screen.getByTestId('test-counter').innerHTML).toBe(value.toString())

    })
})