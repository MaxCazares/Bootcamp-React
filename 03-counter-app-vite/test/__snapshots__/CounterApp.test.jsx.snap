// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2
    data-testid="test-counter"
  >
    100
  </h2>
  <button
    aria-label="btn-decrease"
  >
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
  <button
    aria-label="btn-increase"
  >
    +1
  </button>
</div>
`;
